sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1582,7 +1582,7 @@ class AdaptiveQueryExecSuite
         // Skew join can apply as the repartition is not optimized out.
         assert(smjWithNum.head.isSkewJoin)
         val customReadersWithNum = collect(smjWithNum.head) {
-          case c: CustomShuffleReaderExec if c.hasCoalescedPartition => c
+          case c: CustomShuffleReaderExec => c
```

Comment:
Here, of the 2 CustomShuffleReaderExec nodes one has 
[0, 0, 0, 72, 0] in CustomShuffleReader.child._resultOption
and the resulting partitions are
0 = {CoalescedPartitionSpec@16752} "CoalescedPartitionSpec(3,4,Some(72))"
1 = {CoalescedPartitionSpec@16752} "CoalescedPartitionSpec(3,4,Some(72))"
2 = {CoalescedPartitionSpec@16753} "CoalescedPartitionSpec(4,5,Some(0))"
3 = {CoalescedPartitionSpec@16753} "CoalescedPartitionSpec(4,5,Some(0))"


and the other has [0, 0, 0, 138, 138]
And resulting partitions
0 = {PartialReducerPartitionSpec@16800} "PartialReducerPartitionSpec(3,0,1,72)"
1 = {PartialReducerPartitionSpec@16801} "PartialReducerPartitionSpec(3,1,2,66)"
2 = {PartialReducerPartitionSpec@16802} "PartialReducerPartitionSpec(4,0,1,66)"
3 = {PartialReducerPartitionSpec@16803} "PartialReducerPartitionSpec(4,1,2,72)"

so I don't think you can make it coalesce w/o "disabling" skew mitigation.  In any case since this fixes the definition of "coalesced" I think it's ok to have to adjust a test.